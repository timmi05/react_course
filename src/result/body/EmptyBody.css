.empty-body{
    font-size: 60px;
    height: 10em;
    display: flex;
    align-items: center;
    justify-content: center;
}