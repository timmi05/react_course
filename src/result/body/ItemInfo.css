.item-info-title-year{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-bottom: 5px;
}

.item-info-title{
    font-size: 18px;
}

.item-info-release-year{
    font-size: 14px;
    border: 1px solid #ffffff;
    padding: 2px 10px;
    border-radius: 3px;
}

.item-info-tagline{
    font-size: 14px;
}