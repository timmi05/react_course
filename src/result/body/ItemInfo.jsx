import React from 'react';
import './item_info.css'

const ItemInfo = props => {
     const { title, tagline, release_date } = props;
     return (
          <div className='item-info'>
            <div className='item-info-title-year'>
                <div className='item-info-title'> { title } </div>
                <span className='item-info-release-year'>{ release_date }</span>
            </div>
            <div className='item-info-tagline'>
                { tagline }
             </div>
         </div>
     );
 };

export default ItemInfo