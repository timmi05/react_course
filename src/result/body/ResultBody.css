.result-body{
    background: #232323;
    padding:  60px 50px 1px;
    min-height: 600px;
}
