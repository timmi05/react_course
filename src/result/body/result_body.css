.result-body{
    display: flex;
    flex-direction: wrap;
    background: #232323;
}