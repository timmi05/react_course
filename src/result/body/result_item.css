.result-item{
    margin-bottom: 40px;
}

.result-item-cover{
    height: 500px;
    width: 350px;
    margin-bottom: 20px;
}