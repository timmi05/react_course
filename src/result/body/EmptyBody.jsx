import React from 'react';
import classnames from 'classnames';
import './EmptyBody.css'

export const EmptyBody = props => {
    const composedClassName = classnames(
        props.classesName,
        'empty-body'
      );
    return(
    <div className={ composedClassName }>No films found</div>
    )
};
