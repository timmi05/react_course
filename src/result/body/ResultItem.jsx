import React from 'react';
import { ItemInfo } from './ItemInfo'
// import ItemInfo from './ItemInfo'
import './ResultItem.css'

export const ResultItem = props => {
    const { posterPath, tagline, title, releaseDate } = props;
    return (
         <div className='result-item'>
            <img className='result-item-cover' src={ posterPath } alt={ title }/>
         <ItemInfo
         posterPath = { posterPath }
         tagline = { tagline }
         title = { title }
         releaseDate = { releaseDate }
         />
        </div>
    );
};
