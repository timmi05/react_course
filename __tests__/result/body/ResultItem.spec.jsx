import React from 'react';
import {shallow} from 'enzyme';
import {ResultItem} from '../../../src/result/body/ResultItem';

describe('ResultItem rendering', () => {
    it('renders correctly', () => {
        const component = shallow(<ResultItem/>);
        expect(component).toMatchSnapshot();
    });

    it('simulate click on image', () => {
        const openMovie = jest.fn();
        const component = shallow(<ResultItem
            openMovie={openMovie}
        />);
        component.find('img').simulate('click');
        expect(openMovie).toHaveBeenCalledTimes(1);
    });
});
