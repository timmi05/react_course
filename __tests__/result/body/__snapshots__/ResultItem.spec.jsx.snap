// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem rendering renders correctly 1`] = `
<div
  className="result-item"
>
  <img
    className="result-item-cover"
  />
  <ItemInfo />
</div>
`;
